x-pack/plugins/maps/public/layers/tiled_vector_layer.tsx

```diff
@@ -0,0 +1,233 @@
+    await this._syncSourceFormatters(syncContext, this._source, this._style);
+    await this._syncMVTUrlTemplate(syncContext);
+  }
+
+  _syncSourceBindingWithMb(mbMap: unknown) {
+    // @ts-ignore
+    const mbSource = mbMap.getSource(this.getId());
+    if (!mbSource) {
+      const sourceDataRequest = this.getSourceDataRequest();
+      if (!sourceDataRequest) {
+        // this is possible if the layer was invisible at startup.
+        // the actions will not perform any data=syncing as an optimization when a layer is invisible
+        // when turning the layer back into visible, it's possible the url has not been resovled yet.
+        return;
+      }
+
+      const sourceMeta: {
+        layerName: string;
+        urlTemplate: string;
+        minSourceZoom: number;
+        maxSourceZoom: number;
+      } | null = sourceDataRequest.getData() as {
+        layerName: string;
+        urlTemplate: string;
+        minSourceZoom: number;
+        maxSourceZoom: number;
+      };
+      if (!sourceMeta) {
+        return;
+      }
+
+      const sourceId = this.getId();
+
+      // @ts-ignore
+      mbMap.addSource(sourceId, {
+        type: 'vector',
+        tiles: [sourceMeta.urlTemplate],
+        minzoom: sourceMeta.minSourceZoom,
+        maxzoom: sourceMeta.maxSourceZoom,
+      });
+    }
+  }
+
+  _syncStylePropertiesWithMb(mbMap: unknown) {
+    // @ts-ignore
+    const mbSource = mbMap.getSource(this.getId());
+    if (!mbSource) {
+      return;
+    }
+
+    const sourceDataRequest = this.getSourceDataRequest();
+    if (!sourceDataRequest) {
+      return;
+    }
+    const sourceMeta: {
+      layerName: string;
+    } = sourceDataRequest.getData() as {
+      layerName: string;
+    };
+    const options = { mvtSourceLayer: sourceMeta.layerName };
+
+    this._setMbPointsProperties(mbMap, options);
+    this._setMbLinePolygonProperties(mbMap, options);
+  }
+
+  _requiresPrevSourceCleanup(mbMap: unknown) {
+    // @ts-ignore
+    const mbTileSource = mbMap.getSource(this.getId());
+    if (!mbTileSource) {
+      return false;
+    }
+    const dataRequest = this.getSourceDataRequest();
+    if (!dataRequest) {
+      return false;
+    }
+    const tiledSourceMeta: {
+      urlTemplate: string;
+      minSourceZoom: number;
+      maxSourceZoom: number;
+    } | null = dataRequest.getData() as {
+      urlTemplate: string;
+      minSourceZoom: number;
+      maxSourceZoom: number;
+    };
+    if (
+      mbTileSource.tiles[0] === tiledSourceMeta.urlTemplate &&
+      mbTileSource.minzoom === tiledSourceMeta.minSourceZoom &&
+      mbTileSource.maxzoom === tiledSourceMeta.maxSourceZoom
+    ) {
+      // TileURL and zoom-range captures all the state. If this does not change, no updates are required.
+      return false;
+    }
+
+    return true;
+  }
+
+  syncLayerWithMB(mbMap: unknown) {
+    const requiresCleanup = this._requiresPrevSourceCleanup(mbMap);
+    if (requiresCleanup) {
```

Comment:
This clean-up logic also exists in https://github.com/elastic/kibana/blob/master/x-pack/plugins/maps/public/layers/vector_tile_layer.js#L164. Mind creating a utility function for it to avoid the duplication?